### ISSUE TITLE
Services needed in integration tests should not be manually registered as public

### ISSUE THREAD

**Issue by grossmannmartin** [shopsys/shopsys#907]
### What is happening

It would be best if we wouldn't have to make every service that is taken from DIC in tests public. 

These copy-pasted service definitions can cause a lot of trouble. Either stop using the container directly (can we autowire services in integration tests?) or have all services public in test env by default (maybe via a compiler pass?)...

### Expected result
We don't need to manually add services into services_test.yml.

**Comment by grossmannmartin** [comment]
Seems like we can fix this issue by upgrading to Symfony 4.1 – https://symfony.com/blog/new-in-symfony-4-1-simpler-service-testing

**Comment by simara-svatopluk** [comment]
I don't even understand why do we use the container in tests. We're just hacking tests and using service locator instead of standard dependency injection... Isn't it possible with PHPUnit to just inject the service?

```php
class ArticleFacadeTest {

    private $facade;

    public function __construct(ArticleFacade $facade) {
        $this->facade = $facade;
    }

    // ...
}
```

**Comment by grossmannmartin** [comment]
I don't know if it's by PHPUnit, Symfony or app (I didn't do enough digging), but a parameter in constructor is not working. 

First of all, Functional tests inherit from `PHPUnit\Framework\TestCase` and there is defined constructor `public function __construct($name = null, array $data = [], $dataName = '')`

And as a second, adding something as a parameter of the constructor throws exception `ArgumentCountError: Too few arguments to function ...::__construct(), 0 passed in ... and exactly 1 expected`

**Comment by simara-svatopluk** [comment]
I'm bit annoyed by this issue, so I tried to search a bit... And there maybe is a hack solution that "looks like dependency injection", and should work with private services... I'll try it tomorrow if there is a time

https://github.com/jakzal/phpunit-injector#symfony-container-symfony-34--40

**Comment by simara-svatopluk** [comment]
It works :smiling_imp: https://github.com/shopsys/shopsys/commit/a3f9e1361da5ab299ea232c839c0e19b5c9b9e1a

**Comment by vitek-rostislav** [comment]
Come on stalebot, do not close this! :angry: 

We already have a solution prototype thanks to @simara-svatopluk :heart: 

It should be quite easy to create a proper PR now and get rid of `services_test.yml` once for all. The only question is, does it pay off to work on it or shall we wait for the upgrade on Symfony 4?

**Comment by simara-svatopluk** [comment]
Hi, I can create a PR for you if you wish... but will you merge it to the master? Will You process it?
I'm asking in advance... because if there will be a PR such as https://github.com/shopsys/phpstorm-inspect/pull/11, I don't have a motivation to make the PR ;-)

**Comment by vitek-rostislav** [comment]
Hi Svaťa, well I want it :grin: We will process it, you have my word :wink:

**Comment by simara-svatopluk** [comment]
Ok, I'll do it. Hopefully during next week. Just one question - which branch should I choose for development? master/7.3/8.0/different?

**Comment by pesektomas** [comment]
This issue can be fixed in #1704
